jormungandr/src/explorer/graphql/config_param.rs

```diff
@@ -0,0 +1,525 @@
+        Self { block_0_date: v.0 }
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Enum)]
+pub enum DiscriminationEnum {
+    Production,
+    Test,
+}
+
+#[derive(SimpleObject)]
+pub struct Discrimination {
+    discrimination: DiscriminationEnum,
+}
+
+impl From<&DiscriminationLib> for Discrimination {
+    fn from(v: &DiscriminationLib) -> Self {
+        match v {
+            DiscriminationLib::Production => Self {
+                discrimination: DiscriminationEnum::Production,
+            },
+            DiscriminationLib::Test => Self {
+                discrimination: DiscriminationEnum::Test,
+            },
+        }
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Enum)]
+pub enum ConsensusTypeEnum {
+    Bft,
+    GenesisPraos,
+}
+
+#[derive(SimpleObject)]
+pub struct ConsensusType {
+    consensus_type: ConsensusTypeEnum,
+}
+
+impl From<&ContainerTypeLib> for ConsensusType {
+    fn from(v: &ContainerTypeLib) -> Self {
+        match v {
+            ContainerTypeLib::Bft => Self {
+                consensus_type: ConsensusTypeEnum::Bft,
+            },
+            ContainerTypeLib::GenesisPraos => Self {
+                consensus_type: ConsensusTypeEnum::GenesisPraos,
+            },
+        }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct SlotsPerEpoch {
+    slots_per_epoch: u32,
+}
+
+impl From<&u32> for SlotsPerEpoch {
+    fn from(v: &u32) -> Self {
+        Self {
+            slots_per_epoch: *v,
+        }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct SlotDuration {
+    slot_duration: u8,
+}
+
+impl From<&u8> for SlotDuration {
+    fn from(v: &u8) -> Self {
+        Self { slot_duration: *v }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct EpochStabilityDepth {
+    epoch_stability_depth: u32,
+}
+
+impl From<&u32> for EpochStabilityDepth {
+    fn from(v: &u32) -> Self {
+        Self {
+            epoch_stability_depth: *v,
+        }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct Milli {
+    milli: u64,
+}
+
+impl From<&MilliLib> for Milli {
+    fn from(v: &MilliLib) -> Self {
+        Self {
+            milli: (*v).to_millis(),
+        }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct BlockContentMaxSize {
+    block_content_max_size: u32,
+}
+
+impl From<&u32> for BlockContentMaxSize {
+    fn from(v: &u32) -> Self {
+        Self {
+            block_content_max_size: *v,
+        }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct AddBftLeader {
+    add_bft_leader: BftLeader,
+}
+
+impl From<&BftLeaderId> for AddBftLeader {
+    fn from(v: &BftLeaderId) -> Self {
+        Self {
+            add_bft_leader: v.clone().into(),
+        }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct RemoveBftLeader {
+    remove_bft_leader: BftLeader,
+}
+
+impl From<&BftLeaderId> for RemoveBftLeader {
+    fn from(v: &BftLeaderId) -> Self {
+        Self {
+            remove_bft_leader: v.clone().into(),
+        }
+    }
+}
+
+#[derive(SimpleObject)]
+pub struct LinearFee {
```

Comment:
Hmmm, technically there is already an object for this in the explorer, but is named `FeeSettings` for one reason or another. It's also slightly different because it flattens the structure, but they could be joined I guess.

I could do it myself later, though.

Reply:
What do you think if I will replace the usage of the `FeeSettings` with this `LinearFee` struct ?

Reply:
I'm ok with it, it was called mostly because I was thinking at the time in the possibility of using something else (non linear?) as fees, but now I don't think it makes much sense tbh.